lib/active_merchant/billing/gateways/firstdata_e4.rb

```diff
@@ -275,7 +275,7 @@ def add_network_tokenization_credit_card(xml, credit_card)
       def add_card_authentication_data(xml, options)
         xml.tag! "CAVV", options[:cavv]
         xml.tag! "XID", options[:xid]
-        xml.tag! "Ecommerce_Flag", options[:eci]
+        xml.tag! "Ecommerce_Flag", options[:eci] if options[:eci].present?
```

Comment:
Not sure if this is totally necessary, but I wanted to guard against cases where the node would be set twice. It's really not ideal how there's 3 different places in the  flow that this value can be set.